src/lib/components/CommunitySelectionModal/CommunitySelectionModal.js

```diff
@@ -0,0 +1,100 @@
+// This file is part of React-Invenio-Deposit
+// Copyright (C) 2022 CERN.
+//
+// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component } from 'react';
+import { Header, Button, Modal } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+import { i18next } from '@translations/i18next';
+import { CommunitySelectionSearch } from './CommunitySelectionSearch';
+import { CommunitySelectionFooter } from './CommunitySelectionFooter';
+import { CommunityContext } from './CommunityContext';
+
+export class CommunitySelectionModal extends Component {
+  constructor(props) {
+    super(props);
+
+    const { chosenCommunity } = this.props;
+
+    this.state = {
+      modalOpen: false,
+      localChosenCommunity: chosenCommunity,
+    };
+
+    this.contextValue = {
+      setLocalCommunity: (community) =>
+        this.setState({ localChosenCommunity: community }),
+      chosenCommunity: () => this.state.localChosenCommunity,
```

Comment:
Because this is a function, it would be better to call it `getChosenCommunity` instead!

Reply:
ah yes, makes sense!